File: src/api/events.ts
import { invoke } from "@tauri-apps/api/core";

import type { EventDto, ListEventsQuery } from "@/api/types";

/** Tauri Command: list_events_cmd */
export async function listEvents(query: ListEventsQuery = {}): Promise<EventDto[]> {
  return invoke("list_events_cmd", { query });
}

File: src-tauri/src/commands/events.rs
use tauri::State;

use crate::errors::AppError;
use crate::events::{list_events, EventDto, ListEventsQuery};
use crate::state::AppState;

#[tauri::command]
pub fn list_events_cmd(
    state: State<AppState>,
    query: ListEventsQuery,
) -> Result<Vec<EventDto>, AppError> {
    list_events(&state.db, query)
}
